VAL-189: Add reasons table to email_validations api documentation (#430)
mailgun/documentation: source/api-email-validation.rst

risk                     string       ``high``, ``medium``, ``low`` or ``null`` Depending on the evaluation of all aspects of the given email.
=====================    =========    ============================================================================================================

> Reason Explanationi:

Reviewer: Please correct the typo.
```suggestion
Reason Explanation:
```

=============================    ==========================================================================================================================================================================================
Reason                           Description
=============================    ==========================================================================================================================================================================================

> unknown_provider                 The MX provider is a high risk unknown provider

Reviewer: ```suggestion
unknown_provider                 The MX provider is a high risk unknown provider.
```

> no_mx / No MX host found         The recipient domain does not have a valid MX host (Note this reason will be consolidated to only "no_mx" in the future)

Reviewer: ```suggestion
no_mx / No MX host found         The recipient domain does not have a valid MX host. _Note: this reason will be consolidated to only "no_mx" in the future._
```

> high_risk_domain                 Information obtained about the domain indicates it is high risk to send email to this domain

Reviewer: ```suggestion
high_risk_domain                 Information obtained about the domain indicates it is high risk to send email to.
```

> subdomain_mailer                 The recipient domain is identified to be a subdomain and is not on our exception list. Subdomains are considered to be high risk as many spammers and malicious actors utilize subdomains

Reviewer: ```suggestion
subdomain_mailer                 The recipient domain is identified to be a subdomain and is not on our exception list. Subdomains are considered to be high risk as many spammers and malicious actors utilize them.
```

> immature_domain                  The domain is newly created based on the WHOIS information

Reviewer: ```suggestion
immature_domain                  The domain is newly created based on the WHOIS information.
```

> tld_risk                         The domain has a TLD that has been identified as high risk

Reviewer: ```suggestion
tld_risk                         The domain has a top-level-domain (TLD) that has been identified as high risk.
```

> mailbox_does_not_exist           The mailbox is undeliverable or does not exist

Reviewer: ```suggestion
mailbox_does_not_exist           The mailbox is undeliverable or does not exist.
```

> mailbox_is_disposable_address    The mailbox has been identified to be a disposable address

Reviewer: ```suggestion
mailbox_is_disposable_address    The mailbox has been identified to be a disposable address. Disposable address are temporary, generally one time use, addresses.
```

> mailbox_is_role_address          The mailbox is a role based address, e.g. support@

Reviewer: ```suggestion
mailbox_is_role_address          The mailbox is a role based address (ex. support@..., marketing@...).
```
